pkg/config/types/v1alpha1/machine_config.go

```diff
@@ -107,7 +107,7 @@ func (m *MachineConfig) CertSANs() []string {
 
 // SetCertSANs implements the Configurator interface.
 func (m *MachineConfig) SetCertSANs(sans []string) {
-	m.MachineCertSANs = sans
+	m.MachineCertSANs = append(m.MachineCertSANs, sans...)
```

Comment:
@bradbeam on second thought, I think we need to ensure that `m.MachineCertSANs` is not `nil`, and create an empty slice if it is.